plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataConfig.java

```diff
@@ -77,8 +77,8 @@ public boolean isComplexJoinPushdownEnabled()
         return complexJoinPushdownEnabled;
     }
 
-    @Deprecated
-    @Config("deprecated.join-pushdown.with-expressions")
```

Comment:
This needs approval from @kokosing  per https://github.com/trinodb/trino/pull/27619#discussion_r2611319566